### PR TITLE
Refactor https component

### PR DESCRIPTION
**Pull request by Behn4m**
This Pull Request matches the Https_Request component with the architecture of project by replacing global variables with  Handlers and renaming APIs

### CODE REVIEW

File: components/HttpsComponent/HttpsRequests.c [hunk 1]
```diff
@@ -85,7 +59,7 @@ static void https_get_request(esp_tls_cfg_t cfg, const char *WEB_SERVER_URL, con
         goto exit;
     }
```
Comment (MohammadAbbasi76) [inline components/HttpsComponent/HttpsRequests.c:61]: I think may be  we need async connection ! , at now we have sync connection  and some time happen bug

File: components/HttpsComponent/HttpsRequests.c [hunk 2]
```diff
@@ -121,41 +87,43 @@ static void https_get_request(esp_tls_cfg_t cfg, const char *WEB_SERVER_URL, con
...
-            ESP_LOGE(TAG, "sending data from Https by queue failed !");
-            vTaskDelete(xTaskHandlerHTTPS);
+        buf[len] = '\0'; // Null-terminate the received data
+
+        
+        if (len > 0) {                  // Check if the read operation completed
+            if (xQueueSend(*HttpsRequestsHandler->ResponseBufQueue, buf, portMAX_DELAY) == pdTRUE)    // Data reading completed; process the data here if needed
+            {
+                ESP_LOGI(TAG, "Reading response data finished, data sent by queue!");
+                xSemaphoreGive(*HttpsRequestsHandler->ResponseReadySemaphore);
```
Comment (MohammadAbbasi76) [inline components/HttpsComponent/HttpsRequests.c:115]: needed mechanism for detecting complete response or broken response and , and if it is broken fix it with something like timeout or waiting for getting complete response ...

File: components/HttpsComponent/HttpsRequests.c [hunk 3]
```diff
@@ -85,7 +59,7 @@ static void https_get_request(esp_tls_cfg_t cfg, const char *WEB_SERVER_URL, con
         goto exit;
     }
 
-    if (esp_tls_conn_http_new_sync(WEB_SERVER_URL, &cfg, tls) == 1)
+    if (esp_tls_conn_http_new_sync(WEB_SERVER_URL, &cfg, tls) == true)
```
Comment (MohammadAbbasi76) [inline components/HttpsComponent/HttpsRequests.c:62]: I think may be  we need async connection ! , at now we have sync connection  and some time happen bug

File: components/HttpsComponent/HttpsRequests.c [hunk 4]
```diff
@@ -121,41 +87,43 @@ static void https_get_request(esp_tls_cfg_t cfg, const char *WEB_SERVER_URL, con
...
     ESP_LOGI(TAG, "Reading HTTP response...");
-    do
-    {
+    
+    do {
         len = sizeof(buf) - 1;
         memset(buf, 0x00, sizeof(buf));
-        // vTaskDelay((SEC*5) / portTICK_PERIOD_MS);
+
         ret = esp_tls_conn_read(tls, (char *)buf, len);
```
Comment (MohammadAbbasi76) [inline components/HttpsComponent/HttpsRequests.c:95]: needed mechanism for detecting complete response or broken response and , and if it is broken fix it with something like timeout or waiting for getting complete response ...

File: main/main.c [hunk 5]
```diff
@@ -35,6 +44,13 @@ void app_main(void)
     Spotify_TaskInit(&SpotifyInterfaceHandler);
     // after this semaphore you can use playback command function in every where !
     if (xSemaphoreTake(IsSpotifyAuthorizedSemaphore, portMAX_DELAY) == pdTRUE)
+    {
         Spotify_SendCommand(GetNowPlaying);
+        vTaskDelay((pdMS_TO_TICKS(SEC * 5)));
+        Spotify_SendCommand(Pause);
```
Comment (MohammadAbbasi76) [inline main/main.c:50]: if increase delay between your request , 
image
got it solve .

Comment (MohammadAbbasi76) [reply main/main.c:50]: or add mechanism for preventing two time defining dynamic variable

**Comment by Behn4m** [comment]
this PR closed because of the new PR which is replacing HTTP implementation with client object

**Pull request closed without merging**